File: app/src/main/java/github/hellocsl/cursorwheellayout/adapter/SimpleImageAdapter.java
package github.hellocsl.cursorwheellayout.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.ImageView;

import java.util.List;

import github.hellocsl.cursorwheel.CursorWheelLayout;
import github.hellocsl.cursorwheellayout.R;
import github.hellocsl.cursorwheellayout.data.ImageData;

/**
 * Created by chensuilun on 16/4/24.
 */
public class SimpleImageAdapter implements CursorWheelLayout.CycleWheelAdapter {
    private List<ImageData> mMenuItemDatas;
    private LayoutInflater mLayoutInflater;
    private Context mContext;

    public SimpleImageAdapter(Context context, List<ImageData> menuItemDatas) {
        mContext = context;
        mLayoutInflater = LayoutInflater.from(context);
        mMenuItemDatas = menuItemDatas;
    }

    @Override
    public int getCount() {
        return mMenuItemDatas == null ? 0 : mMenuItemDatas.size();
    }

    @Override
    public View getView(View parent, int position) {
        ImageData item = getItem(position);
        View root = mLayoutInflater.inflate(R.layout.wheel_image_item, null, false);
        ImageView imgView = (ImageView) root.findViewById(R.id.wheel_menu_item_iv);
        imgView.setImageResource(item.mImageRes);
        return root;
    }

    @Override
    public ImageData getItem(int position) {
        return mMenuItemDatas.get(position);
    }

}

File: app/src/main/java/github/hellocsl/cursorwheellayout/adapter/SimpleImageAdapter.kt
package github.hellocsl.cursorwheellayout.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.widget.ImageView
import github.hellocsl.cursorwheel.CursorWheelLayout.CycleWheelAdapter
import github.hellocsl.cursorwheellayout.R
import github.hellocsl.cursorwheellayout.data.ImageData

/**
 * Created by chensuilun on 16/4/24.
 */
class SimpleImageAdapter(
    mContext: Context, private val mMenuItemDatas: List<ImageData>
) : CycleWheelAdapter() {

    private val mLayoutInflater: LayoutInflater = LayoutInflater.from(mContext)
    override fun getCount(): Int {
        return mMenuItemDatas.size
    }

    override fun getView(parent: View?, position: Int): View {
        val (mImageRes) = getItem(position)
        val root = mLayoutInflater.inflate(R.layout.wheel_image_item, null, false)
        val imgView = root.findViewById<ImageView>(R.id.wheel_menu_item_iv)
        imgView.setImageResource(mImageRes)
        return root
    }

    override fun getItem(position: Int): ImageData {
        return mMenuItemDatas[position]
    }
}